### PR TITLE
[Imaging-369] Fix array bounds exception for TIFF with internal JPEG …

### PR DESCRIPTION
**Pull request by gwlucastrig**
This pull request addresses an error introduced by Imaging-194 in which an edge-case could cause an array-bounds exception when reading an odd-sized TIFF file with internal JPEG compression (TIFF with JPEG is a new feature contributed by Imaging-194).

The pull request includes a code fix, expanded JUnit test, and a new test file.  It fixes the problem and slightly improves the JUnit test coverage. Code passed spotbugs and checkstyle checks.

### CODE REVIEW

File: src/test/java/org/apache/commons/imaging/formats/tiff/TiffJpegTest.java [hunk 1]
```diff
@@ -161,6 +166,11 @@ public void testFullPixelMatch() throws IOException {
 
         // test set 2 includes a TIFF-specific alpha channel
         processTestSet(testSet2);
+
+        // test set 3 includes an add-valued width and height that does
+        // not evently fit a block of JPEG data. It is used to test
+        // for the proper handling of edge-cases.
+         processTestSet(testSet3);
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/imaging/formats/tiff/TiffJpegTest.java:173]: oops, looks an extra space.

File: src/test/java/org/apache/commons/imaging/formats/tiff/TiffJpegTest.java [hunk 2]
```diff
@@ -51,6 +51,11 @@ public class TiffJpegTest extends TiffBaseTest {
         "TestSampleArgb.tiff",
         "TestJpegArgb.tiff",};
 
+      static final String[] testSet3 = {
+        "TestSampleRgb127x127.tiff",
+        "TestJpegRgb127x127.tiff",};
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/imaging/formats/tiff/TiffJpegTest.java:56]: Remove trailing comma.

File: src/test/java/org/apache/commons/imaging/formats/tiff/TiffJpegTest.java (hunk 1 above)
Comment (gwlucastrig) [reply src/test/java/org/apache/commons/imaging/formats/tiff/TiffJpegTest.java]: done

File: src/test/java/org/apache/commons/imaging/formats/tiff/TiffJpegTest.java (hunk 2 above)
Comment (gwlucastrig) [reply src/test/java/org/apache/commons/imaging/formats/tiff/TiffJpegTest.java]: done
